source/containers.tex

```diff
@@ -7798,6 +7840,1109 @@
+\begin{itemdecl}
+static constexpr hive_limits block_capacity_default_limits() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A \tcode{hive_limits} struct
+with the \tcode{min} and \tcode{max} members set to
+the implementation's default limits.
+
+\pnum
+\complexity
+Constant.
+\end{itemdescr}
+
+\indexlibrarymember{block_capacity_hard_limits}{hive}%
+\begin{itemdecl}
+static constexpr hive_limits block_capacity_hard_limits() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A \tcode{hive_limits} struct
+with the \tcode{min} and \tcode{max} members set to
+the implementation's hard limits.
+
+\pnum
+\complexity
+Constant.
+\end{itemdescr}
+
+\indexlibrarymember{reshape}{hive}%
+\begin{itemdecl}
+void reshape(hive_limits block_limits);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{T} is \oldconcept{MoveInsertable} into \tcode{hive}.
+
+\pnum
+\effects
+For any active blocks not within the bounds of \tcode{block_limits},
+the elements within those active blocks are reallocated
+to new or existing element blocks which are within the bounds.
+Any element blocks not within the bounds of \tcode{block_limits}
+are deallocated.
+If an exception is thrown during allocation of a new element block,
+\tcode{capacity()} may be reduced,
+reallocation may occur, and
+\exposid{current-limits} may be assigned
+a value other than \tcode{block_limits}.
+Otherwise \tcode{block_limits} is assigned to \exposid{current-limits}.
+If any other exception is thrown the effects are unspecified.
+
+\pnum
+\ensures
+\tcode{size()} is unchanged.
+
+\pnum
+\complexity
+Linear in the number of element blocks in \tcode{*this}.
+If reallocation happens, also linear in the number of elements reallocated.
+
+\pnum
+\remarks
+This operation may change \tcode{capacity()}.
+If reallocation happens, the order of the elements in \tcode{*this} may change.
+Reallocation invalidates all references, pointers, and iterators
+referring to the elements in \tcode{*this},
+as well as the past-the-end iterator.
+\begin{note}
+If no reallocation happens, they remain valid.
+\end{note}
+\end{itemdescr}
+
+\rSec3[hive.modifiers]{Modifiers}
+
+\indexlibrarymember{emplace}{hive}%
+\begin{itemdecl}
+template<class... Args> iterator emplace(Args&&... args);
+template<class... Args> iterator emplace_hint(const_iterator hint, Args&&... args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{T} is \oldconcept{EmplaceConstructible} into \tcode{hive} from \tcode{args}.
+
+\pnum
+\effects
+Inserts an object of type \tcode{T}
+constructed with \tcode{std::forward<Args>(args)...}.
+The \tcode{hint} parameter is ignored.
+If an exception is thrown, there are no effects.
+\begin{note}
+\tcode{args} can directly or indirectly refer to a value in \tcode{*this}.
+\end{note}
+
+\pnum
+\returns
+An iterator that points to the new element.
+
+\pnum
+\complexity
+Constant. Exactly one object of type \tcode{T} is constructed.
```

Comment:
Object construction is not a complexity. It's an effect.

Alternatively, it's an informative explanation of why the complexity is constant, but that makes it a note.

Reply:
This did strike me as odd, but it does seem in line other library wording. [[vector.cons] p10](https://eel.is/c++draft/vector.cons#10) plays fast and loose with what a "Complexity" specification should contain, so it's really not an issue with the paper that would need to be fixed up.